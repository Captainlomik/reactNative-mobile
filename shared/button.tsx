import { Pressable, PressableProps, Text, StyleSheet, Animated, GestureResponderEvent } from "react-native";
import { Colors, Fonts, Radius } from "./tokens";

export function Button({ text, ...props }: PressableProps & { text: string }) {
    const animatedValue = new Animated.Value(100)
    // const color = animatedValue.interpolate({
    // 	inputRange: [0, 100],
    // 	outputRange: ["#45281", "#6C38CC"],
    // });


    const fadeIn = (e: GestureResponderEvent) => {
        Animated.timing(animatedValue, {
            toValue: 0,
            duration: 100,
            useNativeDriver: false
        }).start()
        props.onPressIn && props.onPressIn(e)
    }

    const fadeOut = (e: GestureResponderEvent) => {
        Animated.timing(animatedValue, {
            toValue: 100,
            duration: 100,
            useNativeDriver: false
        }).start()
        props.onPressOut && props.onPressOut(e)
    }

    return (
        <Pressable {...props} onPressIn={fadeIn} onPressOut={fadeOut}>
            <Animated.View style={{
                ...styles.button,
            }}>
                <Text style={styles.text}>{text}</Text>
            </Animated.View>
        </Pressable>
    )
}

const styles = StyleSheet.create({
    button: {
        justifyContent: 'center',
        alignItems: 'center',
        borderRadius: Radius.r10,
        height: 58,
        backgroundColor: Colors.primary,
    },
    text: {
        color: Colors.white,
        fontSize: Fonts.f18,
        fontFamily: 'FiraSans'
    }
})