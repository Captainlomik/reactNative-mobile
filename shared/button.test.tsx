import * as renderer from "react-test-renderer";
import { Button } from "./button";

describe('<Button />', () => {
  it("simple button", () => {
    const tree = renderer.
      create(<Button text="Hello" />).
      toJSON()

    expect(tree).toMatchSnapshot()
  })
  it("simple button", () => {
    const tree = renderer.
      create(<Button text="Hello" isloading={true} />).
      toJSON()

    expect(tree).toMatchSnapshot()
  })
})
