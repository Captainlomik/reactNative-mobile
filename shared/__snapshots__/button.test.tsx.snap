// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> simple button 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
>
  <View
    collapsable={false}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#6C38CC",
        "borderRadius": 10,
        "height": 58,
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        {
          "color": "#FAFAFA",
          "fontFamily": "FiraSans",
          "fontSize": 18,
        }
      }
    >
      Hello
    </Text>
  </View>
</View>
`;

exports[`<Button /> simple button 2`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
>
  <View
    collapsable={false}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#6C38CC",
        "borderRadius": 10,
        "height": 58,
        "justifyContent": "center",
      }
    }
  >
    <ActivityIndicator
      color="#FAFAFA"
      size="large"
    />
  </View>
</View>
`;
